Sources/ResilientDecoding/ResilientPropertyLevelErrors.swift

```diff
@@ -0,0 +1,156 @@
+// Created by George Leontiev on 4/3/20.
+// Copyright © 2020 Airbnb Inc. All rights reserved.
+
+import Foundation
+
+/**
+ The `Resilient` property wrapper has a `projectedValue` in `DEBUG` builds which contains the error(s) encountered when decoding that property. This is intended to only be used during development, error reporting in release should instead be done via `ResilientDecodingErrorReporter`.
+ Limiting this functionality to `DEBUG` allows us to trade some extra complexity in this file for better [_clarity at point of use_](https://swift.org/documentation/api-design-guidelines/) and guarantee that this doesn't impact release builds.
+ */
+#if DEBUG
+
+extension Resilient {
+
+  @dynamicMemberLookup
+  public struct PropertyLevelErrors {
+    let decodedValue: Value
+
+    /**
+     The error encountered while decoding the value
+     */
+    public let error: Error?
+
+    /**
+     This type defines additional ways decoding errors can be inspected on `Resilient<[T]>` and `Resilient<[T]?>`
+     The properties here are meant to be accessed directly on `PropertyLevelErrors` via `@dynamicMemberLookup`
+     */
+    public struct ArrayErrorDigest<Element> {
+      
+      /**
+       An array of `Result` where elements that failed to decode are represented by `failure` case in the position where they would have occurred had the entire array been decoded successully.
+       If the entire resilient array failed to decode, this will be a single failed `Result` with the top-level error.
+       */
+      public let results: [Result<Element, Error>]
+
+      /**
+       All errors encountered when decoding the resilient array.
```

Comment:
Should we clarify that the indexes won't be the same as the element in the array here?

Reply:
I'll add a note